### PR TITLE
Write Divider `SPEC.md`

### PR DESCRIPTION
**Pull request by lawrencewin**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

This PR gets rid of the boilerplate spec from the component generator in the Divider package. Here, I've copied over some documentation from the original spec document, and I've updated some examples to align with other specs and reference existing Divider code.

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by lawrencewin** [comment]
@rurikoaraki Is there anything missing from this spec file? I used RadioGroup and Avatar as examples. I think the only thing I should add are visual examples.

@rohanprasadofficial feel free to add things to the spec file regarding android.

**Comment by rohanprasadofficial** [comment]
@lawrencewin , added a mobile description. Apart from it not supported props are already documented. 
Looks good from my side, only adding visual will do.

File: packages/experimental/Divider/SPEC.md [hunk 1]
```diff
@@ -2,4 +2,161 @@
...
+   * @default false
+   * Note: This prop is not supported on mobile platforms(Android & iOS).
+   */
+  vertical?: boolean;
+}
+```
+
+### Styling Tokens
+
+Tokens can be used to customize the styling of the control by using the customize function on the `Divider`. For more information on using the customize API, please see [this page](https://github.com/microsoft/fluentui-react-native/blob/main/packages/framework/composition/README.md). The `Divider` has the following tokens:
```
Comment (rurikoaraki) [inline packages/experimental/Divider/SPEC.md:124]: The customize API doesn't exist on the compose framework, so users will have to use compose to alter tokens.

Comment (lawrencewin) [reply packages/experimental/Divider/SPEC.md:124]: Is this true? In the test app, I'm somehow using `customize` to change the color tokens on my Divider. 
``` 
const RedDivider = Divider.customize({ contentColor: 'red', lineColor: 'red' });
```

Comment (rurikoaraki) [reply packages/experimental/Divider/SPEC.md:124]: Whoops you're 100% correct, compressible doesn't have compose, but it does have customize. Sorry for the confusion.

File: packages/experimental/Divider/SPEC.md [hunk 2]
```diff
@@ -2,4 +2,161 @@
...
+The `Divider` uses 6 total slots:
+
+- `root` [View] The outer container of the component
+- `beforeLine` [View] The dividing line rendered before passed content. If no content is passed, this is the only slot rendered.
+- `afterLine` [View] The dividing line rendered after passed content. Not rendered if no content is passed.
+- `wrapper` [View] If content is passed, this is the outer container of the content between `beforeLine` and `afterLine` which provides the correct padding for the content.
+- `text` [TextV1] If text is passed as a child, this renders inside `wrapper`.
+- `icon` [IconV1] If the `icon` prop is set, an icon will render inside `wrapper`. If text is also passed while the `icon` prop is set, then this slot will not render.
+
+These slots can be modified using the `compose` function on `Divider`. For more information on using the compose API, please see [this page](https://github.com/microsoft/fluentui-react-native/blob/main/packages/framework/composition/README.md).
```
Comment (rurikoaraki) [inline packages/experimental/Divider/SPEC.md:82]: I think you have to expose the slots as props if you want them to be able to be modified. No compose on compressible, unfortunately

**Review by rurikoaraki — APPROVED** [review]

**Pull request merged**